src/languageServerInstaller.ts

```diff
@@ -120,17 +129,36 @@ export class LanguageServerInstaller {
 				title: "Installing terraform-ls"
 			}, (progress, token) => {
 				token.onCancellationRequested(() => {
+					// TODO clean up partial downloads
 					return reject();
 				});
 
-				progress.report({ increment: 10 });
+				progress.report({ increment: 30 });
+
 				return new Promise<void>((resolve, reject) => {
```

Comment:
Promises chain from in the inside, you can remove some of the "promisification" here and rewrite it as
```
return this.download(...)
    .then(() => this.verify())
    .then(() => this.unpack())
    .then(() => fs.unlinkSync(destination)
```

You don't need to "catch and reject", promises.then that return a promise will resolve and be forwarded to the next `.then()` on the chain.

Javascript is very subjective which is one of the reasons why I switched to Go :p these are just some suggestions that can help reduce some LOC.

Reply:
it's important to not in my syntax the fat arrow func "one liner" means the invoked method has an implicit `return` in front of it, for multiline logic make sure to include the return
```
return this.download(...)
    .then(() => {
        // do stuff
        // do more stuff
        return this.verify();
    });

Reply:
Thanks, I expanded it out because we were having that weird problem where it wouldn't finish the install, so I wanted to make sure I could see how everything was being caught. I'd prefer to condense this.